tt_metal/impl/kernels/kernel.cpp

```diff
@@ -114,13 +114,13 @@ uint8_t ComputeKernel::expected_num_binaries() const {
     return 3;
 }
 
-std::vector<ll_api::memory> const &Kernel::binaries(chip_id_t device_id) const {
+std::vector<ll_api::memory> const &Kernel::binaries(uint32_t harvesting_mask) const {
     int expected_num_binaries = this->expected_num_binaries();
-    if (this->binaries_.find(device_id) != this->binaries_.end() and this->binaries_.at(device_id).size() != expected_num_binaries) {
+    if (this->binaries_.find(harvesting_mask) != this->binaries_.end() and this->binaries_.at(harvesting_mask).size() != expected_num_binaries) {
         TT_THROW("Expected " + std::to_string(expected_num_binaries) + " binaries but have "
-                    + std::to_string(this->binaries_.at(device_id).size()) + " for kernel " + this->name());
+                    + std::to_string(this->binaries_.at(harvesting_mask).size()) + " for kernel " + this->name());
     }
-    return this->binaries_.at(device_id);
+    return this->binaries_.at(harvesting_mask);
 }
```

Comment:
Maybe I'm missing something, but kernels belong to a program and programs don't span devices. Why do we have a device_id/harvesting_mask map here?

Reply:
kernels can be compiled for multiple devices, so one kernel can have multiple instances, need to grab the right one.  and, this is a dependency that we need to break (fix that is) - should only have one instance of a kernel that is device agnostic